src/components/Search/CompareWithBase.tsx

```diff
@@ -99,7 +99,7 @@ function CompareWithBase({
     (baseRepository !== 'try' && newRepository === 'try');
 
   const possiblyPreventFormSubmission = (e: React.FormEvent) => {
-    const isFormReadyToBeSubmitted = baseRevs.length > 0;
+    const isFormReadyToBeSubmitted = baseInProgressRevs.length > 0;
```

Comment:
Great questions!

> The "staging" states in SearchView are pretty much useless, right?

Yes, because there's no edit mode, so that's just "in progress" really. There's even no "commited" state because when clicking "compare" the user goes to the compare results page.

> We only use the "staging" states in regards to the "Cancel" action.

Yes that's true!

> Why not use the "commited" state values instead of the "staging" state values to overwrite the "inprogress" state when the user presses the "Cancel" button? This way, I don't think we need the "staging" states. What do you think?

Think of these steps:

1. The user presses edit.
2. They change the revisions.
3. They press "save".
4. They press edit again.
5. They change the revisions.
6. They press "cancel".

=> What revisions should be displayed? What does the user expect? The ones from before step 1, or the ones from after step 3?
I think (in the current situation) that it should the ones after step 3. That's why we have the staging state.
Does that make sense?

_But_ we seem to agree that we don't need all this complexity and we'll get rid of this "save" and this staging state anyway :-)